includes/class-debug.php

```diff
@@ -58,29 +66,103 @@ public static function log( $message, $type = 'string' ) {
 				error_log( 'AspireUpdate - Could not open or write to the file system. Check file system permissions to debug log directory.' );
 				// phpcs:enable
 			}
-			return;
+			return false;
 		}
+		return true;
+	}
 
-		$timestamp         = gmdate( 'Y-m-d H:i:s' );
-		$formatted_message = sprintf(
-			"[%s] [%s]: %s\n" . PHP_EOL,
-			$timestamp,
-			strtoupper( $type ),
-			self::format_message( $message )
-		);
+	/**
+	 * Get the content of the log file truncated upto N number of lines.
+	 *
+	 * @param integer $limit Max no of lines to return. Defaults to a 1000 lines.
+	 *
+	 * @return string The File content truncate upto the number of lines set in the limit parameter.
```

Comment:
Looking at this again, this method is written with the view log use case in mind. That's fine, though I think we can improve on this a bit.

1. The return value is escaped and translated.
    - This prevents potential processing before output. Sanitize Early, Escape Late (as possible)
    - Instead, the escaping can be done before the JSON response is set.
2. The return value is a string whether it's successful or not.
    - This makes it more difficult for the receiving code to determine _how_ the value should be used or presented.
    - If this is preferred to be a _pretty_ low-level method, we could return `string|false` and let the receiving code determine the message on failure.
    - The method determines the return value for an empty log file. Maybe it should just return an empty string and the receiving code can determine how to communicate that?
    - If this is preferred to be a high-level method, we can return `new WP_Error( $err_code, $err_message );` so errors can be identified and context can be understood. The receiving code can then check `is_wp_error( $return_value )` and use `return_value->get_error_message()`, appropriately escaping before output.

Reply:
I guess we should go the WP error class way for future benefits.  Will update the PR

Reply:
Cool! When I get a chance, I'll take a look at the rest of the codebase to see if there's opportunities with other return values and open an issue if I find any, though I don't have any in mind at the moment.